conans/client/tools/intel.py

```diff
@@ -0,0 +1,108 @@
+import os
+import platform
+from contextlib import contextmanager
+
+from conans.client.tools.env import environment_append, env_diff
+from conans.client.tools.win import is_win64, _system_registry_key, MSVS_YEAR
+from conans.errors import ConanException
+from conans.util.env_reader import get_env
+
+
+# https://software.intel.com/en-us/articles/intel-compiler-and-composer-update-version-numbers-to-compiler-version-number-mapping
+INTEL_YEAR = {"19.1": "2020",
+              "19": "2019",
+              "18": "2018",
+              "17": "2017",
+              "16": "2016",
+              "15": "2015"}
+
+
+def intel_installation_path(version, arch):
+    installation_path = get_env("CONAN_INTEL_INSTALLATION_PATH")
+    if installation_path:
+        return installation_path
+
+    system = platform.system()
+    if system in ["Linux", "Darwin"]:
+        subdir = "mac" if system == "Darwin" else "linux"
+        year = INTEL_YEAR.get(version)
+        installation_path = os.path.join(os.sep, "opt", "intel",
+                                         "compilers_and_libraries_%s" % year, subdir)
+    elif system == "Windows":
+        if arch == "x86":
+            intel_arch = "IA32"
+        elif arch == "x86_64":
+            intel_arch = "EM64T"
+        else:
+            raise ConanException("don't know how to find Intel compiler on %s" % arch)
+        if is_win64():
+            base = r"SOFTWARE\WOW6432Node\Intel\Suites\{version}".format(version=version)
+        else:
+            base = r"SOFTWARE\Intel\Suites\{version}".format(version=version)
+        from six.moves import winreg  # @UnresolvedImport
+        path = base + r"\Defaults\C++\{arch}".format(arch=intel_arch)
+        subkey = _system_registry_key(winreg.HKEY_LOCAL_MACHINE, path, "SubKey")
+        if not subkey:
+            raise ConanException("unable to find Intel C++ compiler installation")
+        path = base + r"\{subkey}\C++".format(subkey=subkey)
+        installation_path = _system_registry_key(winreg.HKEY_LOCAL_MACHINE, path, "LatestDir")
+        if not installation_path:
+            raise ConanException("unable to find Intel C++ compiler installation")
+    else:
+        raise ConanException("don't know how to find Intel compiler on %s" % system)
+    return installation_path
+
+
+def compilervars_command(settings, arch=None, compiler_version=None, force=False):
```

Comment:
In the same way that VS use the ``vcvars`` naming, why is this generic naming? Shouldn't it be ``intel_vars_command`` or similar?

Reply:
`vcvars.bat` and `compilervars.bat` (`compilervars.sh`) are names of the scripts. also, as the function is already in intel module, it would be an unnecessary repetition of `intel` word.

Reply:
Ok, understood, good point.